File: app/src/main/java/cat/mvmike/minimalcalendarwidget/infrastructure/PermissionService.java
// Copyright (c) 2020, Miquel Martí <miquelmarti111@gmail.com>
// See LICENSE for licensing information

package cat.mvmike.minimalcalendarwidget.infrastructure;

import android.Manifest;
import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;
import androidx.core.content.ContextCompat;

import cat.mvmike.minimalcalendarwidget.application.activity.PermissionsActivity;

public final class PermissionService {

    public static boolean hasPermissions(final Context context) {
        return ContextCompat.checkSelfPermission(context, Manifest.permission.READ_CALENDAR) == PackageManager.PERMISSION_GRANTED;
    }

    public static void launchPermissionsActivity(final Context context) {

        Intent permissionIntent = new Intent(context, PermissionsActivity.class).addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        context.startActivity(permissionIntent);
    }

}

File: app/src/main/kotlin/cat/mvmike/minimalcalendarwidget/infrastructure/permission/PermissionService.kt
// Copyright (c) 2016, Miquel Martí <miquelmarti111@gmail.com>
// See LICENSE for licensing information
package cat.mvmike.minimalcalendarwidget.infrastructure.permission

import android.Manifest.permission.READ_CALENDAR
import android.content.Context
import android.content.Intent
import android.content.pm.PackageManager.PERMISSION_GRANTED
import androidx.core.content.ContextCompat.checkSelfPermission
import cat.mvmike.minimalcalendarwidget.infrastructure.activity.PermissionsActivity

object PermissionService {

    fun hasPermissions(context: Context) = checkSelfPermission(context, READ_CALENDAR) == PERMISSION_GRANTED

    fun launchPermissionsActivity(context: Context) =
        context.startActivity(
            Intent(context, PermissionsActivity::class.java).addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        )
}
